src/wallet/wallet.cpp

```diff
@@ -2857,9 +2862,9 @@ bool CWallet::CreateTransactionInternal(
 
                 // Never create dust outputs; if we would, just
                 // add the dust to the fee.
-                // When nChange is less than the cost of the change output,
-                // send it to fees (this means BnB was used)
-                if (IsDust(newTxOut, discard_rate) || nChange <= cost_of_change)
+                // When the selected_eff is within the exact match range
+                // (nValue + not_input_fees + cost_of_change), don't make change.
+                if (IsDust(newTxOut, discard_rate) || selected_eff <= nValue + not_input_fees + cost_of_change)
```

Comment:
Didn't we just call the first two of that triplet `actual_target`?

Reply:
It's named that in `SelectCoins`. I believe we discussed changing this, but I decided to leave that for a followup refactor that cleans up this whole function.